camel-idea-plugin/src/main/java/com/github/cameltooling/idea/completion/header/CamelHeaderEndpointSource.java

```diff
@@ -22,71 +6,96 @@
 import com.intellij.openapi.util.text.StringUtil;
 import com.intellij.psi.PsiElement;
 import org.jetbrains.annotations.NotNull;
 
-/**
- * Defines all the supported source of endpoints.
- */
 public enum CamelHeaderEndpointSource {
-    /**
-     * Represents the source that provides all the producer endpoints.
-     */
     PRODUCER_ONLY {
         @Override
         Collection<CamelHeaderEndpoint> getEndpoints(@NotNull PsiElement element) {
-            final CamelIdeaUtils utils = CamelIdeaUtils.getService();
-            return utils.findEndpointUsages(
-                ModuleUtilCore.findModuleForPsiElement(element), e -> e.indexOf(':') != -1
-            )
-                .stream()
-                .filter(utils::isProducerEndpoint)
-                .map(PsiElement::getText)
-                .map(StringUtil::unquoteString)
-                .map(StringUtils::asComponentName)
-                .map(CamelHeaderEndpoint::producerOnly)
-                .collect(Collectors.toSet());
+            return new ProducerEndpointFinder(element).findEndpoints();
         }
     },
-    /**
-     * Represents the source that provides all the consumer endpoints.
-     */
+
     CONSUMER_ONLY {
         @Override
         Collection<CamelHeaderEndpoint> getEndpoints(@NotNull PsiElement element) {
-            final CamelIdeaUtils utils = CamelIdeaUtils.getService();
-            return utils.findEndpointDeclarations(
-                ModuleUtilCore.findModuleForPsiElement(element), e -> e.indexOf(':') != -1
-            )
-                .stream()
-                .filter(utils::isConsumerEndpoint)
-                .map(PsiElement::getText)
-                .map(StringUtil::unquoteString)
-                .map(StringUtils::asComponentName)
-                .map(CamelHeaderEndpoint::consumerOnly)
-                .collect(Collectors.toSet());
+            return new ConsumerEndpointFinder(element).findEndpoints();
         }
     },
-    /**
-     * Represents the source that provides all the existing endpoints.
-     */
+
     ALL {
         @Override
         Collection<CamelHeaderEndpoint> getEndpoints(@NotNull PsiElement element) {
-            return Stream.concat(
-                CONSUMER_ONLY.getEndpoints(element).stream(), PRODUCER_ONLY.getEndpoints(element).stream()
-            )
-                .map(CamelHeaderEndpoint::getComponentName)
-                .map(CamelHeaderEndpoint::both)
-                .collect(Collectors.toSet());
+            final Collection<CamelHeaderEndpoint> consumerEndpoints = CONSUMER_ONLY.getEndpoints(element);
+            final Collection<CamelHeaderEndpoint> producerEndpoints = PRODUCER_ONLY.getEndpoints(element);
+
+            return Stream.concat(consumerEndpoints.stream(), producerEndpoints.stream())
+                    .map(CamelHeaderEndpoint::getComponentName)
+                    .map(CamelHeaderEndpoint::both)
+                    .collect(Collectors.toSet());
```

Comment:
ditto

Reply:
It's because of additional space just dorry for that